spec/controllers/settings_controller_spec.rb

```diff
@@ -1,5 +1,29 @@
 require 'spec_helper'
 
 describe SettingsController, type: :controller do
+  let(:user) { create(:user) }
+  let(:attributes) { attributes_for :user }
+
+  
+  before do
+    allow(self.controller).to receive(:current_user).and_return(user)
```

Comment:
Feel free to use the login helper at https://github.com/benwoody/shrtnr/blob/master/spec/support/authentication.rb#L3